core/src/nlp/mod.rs

```diff
@@ -83,23 +91,52 @@ impl Default for ChatAgentConfig {
             llm_uri: "http://localhost:11434".to_string(),
             llm_model: "llama2".into(),
             // mcp_uri: "http://localhost:8080".to_string(),
+            system_prompt: PathBuf::from("./prompts/ultron.md"),
         }
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct ChatHistory(Arc<RwLock<Vec<Event>>>);
+
+impl ChatHistory {
+    pub fn new(initial_history: impl IntoIterator<Item = Event>) -> Self {
+        let chat_history = initial_history.into_iter().collect();
+        Self(Arc::new(RwLock::new(chat_history)))
+    }
+
+    pub async fn append(&self, event: Event) {
+        let mut history = self.0.write().await;
+        history.extend([event]);
+    }
+
+    /// return a read-only snapshot of the chat history
+    #[instrument(skip(self))]
+    pub async fn read(&self) -> Vec<Event> {
```

Comment:
we have to clone here since there's no way to keep these references around without hogging the lock or some other trickery (like `papaya` or `dashmap`).